webpack.config.js

```diff
@@ -28,42 +28,69 @@ const commonEntryFiles = {
     background: [path.resolve(__dirname, 'src/background/background-init.ts')],
 };
 
-const commonConfig = {
-    entry: commonEntryFiles,
-    module: {
-        rules: [
-            {
-                test: /\.tsx?$/,
-                use: [
-                    {
-                        loader: 'ts-loader',
-                        options: {
-                            transpileOnly: true,
-                            experimentalWatchApi: true,
+const electronEntryFiles = {
+    main: [path.resolve(__dirname, 'src/electron/main/main.ts')],
+    injected: [path.resolve(__dirname, 'src/injected/stylesheet-init.ts'), path.resolve(__dirname, 'src/injected/client-init.ts')],
+};
+
+const createCommonConfig = entry => {
+    const baseConfig = {
+        entry,
+        module: {
+            rules: [
+                {
+                    test: /\.tsx?$/,
+                    use: [
+                        {
+                            loader: 'ts-loader',
+                            options: {
+                                transpileOnly: true,
+                                experimentalWatchApi: true,
+                            },
                         },
-                    },
-                ],
-                exclude: ['/node_modules/'],
-            },
-        ],
-    },
-    resolve: {
-        modules: [path.resolve(__dirname, 'node_modules')],
-        extensions: ['.tsx', '.ts', '.js'],
+                    ],
+                    exclude: ['/node_modules/'],
+                },
+            ],
+        },
+        resolve: {
+            modules: [path.resolve(__dirname, 'node_modules')],
+            extensions: ['.tsx', '.ts', '.js'],
+        },
+        plugins: commonPlugins,
+        node: {
+            setImmediate: false,
+        },
+        performance: {
+            // We allow higher-than-normal sizes because our users only have to do local fetches of our bundles
+            maxEntrypointSize: 10 * 1024 * 1024,
+            maxAssetSize: 10 * 1024 * 1024,
+        },
+    };
+
+    return baseConfig;
+};
+
+const electronConfig = {
+    ...createCommonConfig(electronEntryFiles),
+    name: 'electron',
+    mode: 'development',
+    output: {
+        path: path.join(__dirname, 'extension/electronBundle'),
+        filename: '[name].bundle.js',
     },
-    plugins: commonPlugins,
     node: {
-        setImmediate: false,
+        __dirname: false,
```

Comment:
Echoing Dan's comment from #686 :
This value for `node` as an object containing `__dirname` and `__filename` properties overrides the `node` value containing a `setImmediate` property returned by `createCommonConfig`. Is that expected?

Reply:
Let me check on this and update you.

Reply:
So, it doesn't impact the process in electron but looks like according to https://webpack.js.org/configuration/node/#nodesetimmediate putting it to `false` might be better for us in terms of webpack performance; since we dont need any global node polyfills

Reply:
So changed it to include that property